packages/flutter_nekoton_bridge/lib/nekoton/helpers/abi.dart

```diff
@@ -270,35 +270,117 @@ Future<PublicKey> extractPublicKey(String boc) async {
+Future<(String, String)> codeToTvc(String code) async {
+  final data = await createLib().codeToTvc(code: code);
+
+  return (data[0], data[1]);
 }
 
-/// Merge code and data to tvc base64 string and return it or throw error
-Future<String> mergeTvc({
+/// Merge code and data to tvc base64 string and return (tvc, hash)
+/// or throw error
+Future<(String, String)> mergeTvc({
   required String code,
   required String data,
-}) {
-  return createLib().mergeTvc(code: code, data: data);
+}) async {
+  final res = await createLib().mergeTvc(code: code, data: data);
+
+  return (res[0], res[1]);
 }
 
 /// Split base64 tvc string into data and code.
-/// Return Tuple2[data, code] or throw error
-Future<Tuple2<String?, String?>> splitTvc(String tvc) async {
+/// Return (data, code) or throw error
+Future<(String?, String?)> splitTvc(String tvc) async {
   final res = await createLib().splitTvc(tvc: tvc);
-  return Tuple2(res[0], res[1]);
+  return (res[0], res[1]);
 }
 
-/// Set salt to code and return base64-encoded string or throw error
-Future<String> setCodeSalt({
+/// Set salt to code and return (tvc, hash) or throw error
+Future<(String, String)> setCodeSalt({
   required String code,
   required String salt,
-}) {
-  return createLib().setCodeSalt(code: code, salt: salt);
+}) async {
+  final data = await createLib().setCodeSalt(code: code, salt: salt);
+
+  return (data[0], data[1]);
 }
 
 /// Get salt from code if possible and return base64-encoded salt or throw error
 Future<String?> getCodeSalt(String code) {
   return createLib().getCodeSalt(code: code);
 }
+
+/// None code-related exception that means that [executeLocal] finished with
+/// specified [errorCode].
+class ExecuteLocalException implements Exception {
+  const ExecuteLocalException(this.errorCode);
+
+  final String errorCode;
+}
+
+/// Run contract locally.
+/// [config] - value from [Transport.getBlockchainConfig]
+///
+/// Returns [boc, transaction] if everything is ok or
+/// throws [ExecuteLocalException] if transaction failed, this is not
+///   code-related problem.
+/// or throws error
+Future<(String, Transaction)> executeLocal({
+  required String config,
+  required String account,
+  required String message,
+  required DateTime utime,
+  required bool disableSignatureCheck,
+  BigInt? overwriteBalance,
+  int? globalId,
+}) async {
+  final result = await createLib().executeLocal(
+    account: account,
+    message: message,
+    utime: utime.millisecondsSinceEpoch,
+    globalId: globalId,
+    config: config,
+    overwriteBalance: overwriteBalance?.toString(),
+    disableSignatureCheck: disableSignatureCheck,
+  );
+  if (result.length == 1) {
+    throw ExecuteLocalException(result[0]);
+  }
+
+  return (result[0], Transaction.fromJson(jsonDecode(result[1])));
+}
+
+/// Unpack data from [contractAbi].
+///Returns optional public key and json-encoded Map<String, Token>
+///or throws error.
```

Comment:
Just a formatting: absent space between /// and text.

Reply:
-